src/helpers/chain.js

```diff
@@ -0,0 +1,582 @@
+  for (let i = 0; i < offers.length; i++) {
+
+    // Get current offer
+    let currentOffer = offers[i]
+
+    // If a price invitation and for the correct feed
+    let invitationType = currentOffer["status"]["invitationSpec"][
+    "invitationMakerName"];
+    let previousOffer = currentOffer["status"]["invitationSpec"][
+    "previousOffer"];
+    if (
+        invitationType === "PushPrice" &&
+        previousOffer === feedOfferId
+    ) {
+      let offerRound = Number(
+        currentOffer["status"]["invitationSpec"]["invitationArgs"][0]["roundId"]
+      );
+
+      /**
+       * If it is an offer for the round we are checking for and there is no
+       * error
+       */
+      if (
+        offerRound === roundId &&
+        !currentOffer["status"].hasOwnProperty("error")
+      ) {
+        return true;
+      } else if (
+        /**
+         * If the currentOffer has no error and the offerRound is less than the
+         * roundId, we return false since no new submissions for future rounds
+         * can be created before completing the current round
+         */
+        offerRound < roundId &&
+        !currentOffer["status"].hasOwnProperty("error")
+      ) {
+        return false;
+      }
+    }
+  }
+  return false;
+};
+
+/**
+ * Function to query price from chain
+ * @param {string} feed feed name of the price to query (Ex. ATOM-USD)
+ * @returns {number} the latest price
+ */
+export const queryPrice = async (feed) => {
+  try {
+    // Read value from vstorage
+    const capDataStr = await readVStorage(feed, false);
+
+    let capData;
+
+    try {
+      //parse the value
+      capData = JSON.parse(JSON.parse(capDataStr).value);
+      capData = JSON.parse(capData.values[0]);
+      // Replace any extra characters
+      capData = JSON.parse(capData.body.replaceAll("\\", ""));
+    } catch (err) {
+      throw new Error("Failed to parse CapData for queryPrice");
+    }
+
+    // Get the latest price by dividing amountOut by amountIn
+    let latestPrice =
+      Number(capData.amountOut.value.digits) /
+      Number(capData.amountIn.value.digits);
+
+    logger.info(feed + " Price Query: " + String(latestPrice));
+    return latestPrice;
+  } catch (err) {
+    logger.error("ERROR querying price: " + err);
+    return -1;
+  }
+};
+
+/**
+ * Function to get oracles feed invitations
+ * @param {string} oracle address of the oracle
+ * @returns {object} an object containing feed invitation IDs. Each field in
+ *                   the object represents the feed name (Ex. ATOM-USD) and its
+ *                   value is a number which is the invitation ID.
+ */
+export const getOraclesInvitations = async (oracle) => {
+  let { agoricNames, fromBoard, vstorage } = await makeRpcUtils({ fetch });
+
+  let feedBoards = agoricNames.reverse;
+
+  let feedInvs = {};
+
+  const current = await getCurrent(String(oracle), fromBoard, { vstorage });
+  const invitations = current.offerToUsedInvitation;
+
+  // For each invitation
```

Comment:
For each invitation what?

Reply:
Fixed in https://github.com/jacquesvcritien/agoric-cl-middleware/commit/29244ced3c6d34785c4643507480179d98d7d8c6